eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -237,6 +237,33 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
       log.debug("restoring channel")
       context.system.eventStream.publish(ChannelRestored(self, data.channelId, peer, remoteNodeId, data))
       txPublisher ! SetChannelId(remoteNodeId, data.channelId)
+
+      // we watch all unconfirmed funding txs, whatever our state is
+      // (there can be multiple funding txs due to rbf, and they can be unconfirmed in any state due to zero-conf)
+      data.metaCommitments.commitments.foreach { commitment =>
+        commitment.localFundingStatus match {
+          case LocalFundingStatus.UnknownFundingTx =>
+            // Legacy single-funded channels. The funding tx may or may not be confirmed, a watch-confirmed will be set
+            // at reconnection if necessary
+            watchFundingSpent(commitment)
+          case _: LocalFundingStatus.SingleFundedUnconfirmedFundingTx =>
+            // A watch-confirmed will be set at reconnection if necessary 
+            watchFundingSpent(commitment)
+          case fundingTx: LocalFundingStatus.DualFundedUnconfirmedFundingTx =>
+            watchFundingConfirmed(fundingTx.sharedTx.txId, fundingTx.fundingParams.minDepth_opt)
+          case fundingTx: LocalFundingStatus.PublishedFundingTx =>
+            // those are zero-conf channels, the min-depth isn't critical, we use the default
+            val fundingMinDepth = nodeParams.channelConf.minDepthBlocks.toLong
+            blockchain ! WatchFundingConfirmed(self, fundingTx.tx.txid, fundingMinDepth)
```

Comment:
It would be a bit more consistent to also call `watchFundingConfirmed`:

```suggestion
            val fundingMinDepth = nodeParams.channelConf.minDepthBlocks.toLong
            watchFundingConfirmed(fundingTx.tx.txid, Some(fundingMinDepth))
```

Reply:
Done with 79970b5ba.